modules/gateways/veritrans.php

```diff
@@ -349,21 +349,65 @@ function veritrans_link($params)
         $params['credit_card']['save_card'] = true;
     }
 
-    // Get snap token
     try {
-        // debug
-        // echo "<pre>";
-        // var_dump($params);
-        // echo "</pre>";
-        // exit();
-        $snap_transaction = Veritrans_Snap::createTransaction($params);
-        $snapToken = $snap_transaction->token;
-        $redirect_url = $snap_transaction->redirect_url;
-        // error_log(" ############# TOKEN ::: ".$snapToken);
+        $whmcsApiCommand = 'GetInvoice';
+        $whmcsApiPostParams = array(
+            'invoiceid' => $invoiceId
+        );
+        $invoice = localAPI($whmcsApiCommand, $whmcsApiPostParams);
+        $existingInvoiceNotes = $invoice->notes;
     } catch (Exception $e) {
-        // error_log('Caught exception: ',  $e->getMessage(), "\n");
-        if (preg_match('/utilized/',$e->getMessage()) ){
-            return "<h4> Awaiting Your Payment </h4>";
+        $existingInvoiceNotes = "";
+    }
+
+    $midtransNotesSeparatorBegin = "#PayUrl: ";
+    $midtransNotesSeparatorEnd = " PayUrl#";
+
+    // Check if Snap URL/token found in invoices notes
+    if (strpos($existingInvoiceNotes, $midtransNotesSeparatorBegin)) {
+        // use Regexp to extract existing Snap Url value between Midtrans invoice notes begin & end separator
+        $midtransNotesRegexPattern = "/$midtransNotesSeparatorBegin(.*?)$midtransNotesSeparatorEnd/";
+        if ( preg_match($midtransNotesRegexPattern, $existingInvoiceNotes, $matches) ) {
+            $redirect_url = $matches[1];
+        }
+        // get Snap Token from string after the last `/` char of $redirect_url
+        $SnapUrlExplodeResults = explode('/', $redirect_url);
+        $snapToken = end( $SnapUrlExplodeResults );
```

Comment:
Sorry I just realized I had a typo on my example code, let's not use capital to start a var name, change `$SnapUrlExplodeResults` to `$snapUrlExplodeResults`

Reply:
Noted for this kak, i find version number in this repo cannot find it. I check on slack mid github monitoring too, no history version for WHMCS plugin ka

Thank you